openedx/core/djangoapps/user_api/accounts/api.py

```diff
@@ -182,6 +199,12 @@ def update_account_settings(requesting_user, update, username=None):
         for serializer in user_serializer, legacy_profile_serializer:
             serializer.save()
 
+        if "account_privacy" in update:
+            if account_privacy is not None:
+                account_privacy_serializer.save()
+            else:
+                delete_user_preference(requesting_user, "account_privacy")
+
```

Comment:
Instead of duplicating the business logic with updating user_preferences here, why not just call `update_user_preferences` in `preferences/api.py`?  It does all that's needed: creates the serializer, validates the serializer, and deletes the preference when needed.